static/js/plotjs/download.js

```diff
@@ -0,0 +1,53 @@
+/* eslint-disable */
+function table_to_csv(source, write_header) {
+  const columns = ['mjd', 'filter', 'flux', 'fluxerr', 'zp',
+      'zpsys', 'lim_mag', 'stacked'];
+  const nrows = source.get_length();
+  const lines = [];
+
+  if (write_header) {
+    lines.push(columns.join(','));
+  }
+
+  for (let i = 0; i < nrows; i++) {
+    const row = [];
+    for (let j = 0; j < columns.length; j++) {
+      const column = columns[j];
+      try {
+        row.push(source.data[column][i].toString());
+      } catch (error) {
+        if (column === 'zp') {
+          row.push(25.0);
+        } else if (column === 'zpsys') {
+          row.push('ab');
+        } else {
+          throw error;
+        }
+      }
+    }
+    lines.push(row.join(','));
+  }
+
+  return lines.join('\n').concat('\n');
+}
+
+
+const filename = 'objname.csv';
```

Comment:
can this file name be dynamically created based on source position or source name? I'm worried about having a bunch of csv files with named `objname (N).csv` in the Downloads folder.

Reply:
This string is just a placeholder and is automatically replaced by the object name in python